src/monitoring_service/service.py

```diff
@@ -82,9 +104,24 @@ def __init__(
             monitoring_service_contract=monitoring_contract,
         )
 
-    def start(self, wait_function: Callable = time.sleep) -> None:
+    def start(
+        self,
+        wait_function: Callable = time.sleep,
+        check_account_gas_reserve: bool = True,
+    ) -> None:
         while True:
             last_confirmed_block = self.web3.eth.blockNumber - self.required_confirmations
+
+            # check gas reserve
+            do_gas_reserve_check = (
+                check_account_gas_reserve and
+                last_confirmed_block % DEFAULT_GAS_CHECK_BLOCKS == 0 and
```

Comment:
We can process multiple blocks at once, so this might be skipped. Why not use
```
last_confirmed_block >= self.last_gas_check_block + DEFAULT_GAS_CHECK_BLOCKS
```